keras/utils/torch_utils.py

```diff
@@ -104,3 +112,36 @@ def _track_module_parameters(self):
 
     def call(self, *args, **kwargs):
         return self.module.forward(*args, **kwargs)
+
+    def save_own_variables(self, store):
+        """Saves model's state from `state_dict`.
+        `model.parameters` excludes some of model's state like
+        `BatchNorm` mean and variance. So, use `state_dict` to obtain
+        all of model's state.
+        """
+        state_dict = self.module.state_dict()
+        # Save all layer's state keys
+        store[self.name + "._keys"] = list(state_dict.keys())
```

Comment:
I don't understand why you need to store the list of keys -- the `store` object is already a dict-like and you should be able to query `.keys()` on it. So just doing 

```python
def save_own_variables(self, store):
    state_dict = self.module.state_dict()
    for key in state_dict.keys():
        store[key] = convert_to_numpy(state_dict[key])
```
should be equivalent

Reply:
There could be multiple `TorchModuleWrapper` in the `Model`. I wanted to identify all the keys that belonged to the specific `TorchModuleWrapper` Layer, hence wanted to store the `keys` that belong to the layer in `store[self.name + '._keys']`.

Reply:
This is already taken into by the serialization logic. Just using keys that are local to the layer is enough.

Reply:
`state_dict` includes keys for `BatchNorm` `mean` and `variance` that are not part of `model.parameters`, and hence missed by default save logic of layer variables.  So the logic was intended to capture all of the `state_dict` keys.

Reply:
@fchollet - Thanks for your patience. I didn't have to save the `_keys` or force a `name`.  It works with just storing and loading from `state_dict`.  Tested with multiple wrappers and batch norm.